### ISSUE TITLE
Require drill down capabilities for buckets

### ISSUE THREAD

**Issue by andyvandesand** [TheAxelander/OpenBudgeteer#167]
Hi,

I'd suggest some kind of drill-down option for each ~~bucket category & bucket per month~~ category in the header. I've loaded all my data, but the income doesn't add up, and nor do any of the other numbers in the header of the bucket page. I'm currently trying to go through the code to work out the specific calculations, but that's obviously quite tedious. Might be nice to publish these details in the docs?

Cheers
Andy

**Comment by TheAxelander** [comment]
Did you assigned the Buckets to all your Transactions that you have loaded? Income Transactions have to be assigned to the pre-defined Bucket `Income`.

Not sure what you mean with drill-down. Something like this (showing Input and Output of a specific Bucket):

![image](https://github.com/TheAxelander/OpenBudgeteer/assets/17849665/bafafe5b-438d-4d18-9983-31d3078ae475)

**Comment by andyvandesand** [comment]
Yes, all assigned, although still trying to get all transfers to add to 0. 

Yes that screen is great, but for e.g. income up the top - currently not adding up, out by quite a significant amount, and there are only 5x transactions marked as income. But all the other categories at the top would be great to have some kind of explanation of how they are calculated, I'm just trying to work out what goes where but turns out it's a lot harder than anticipated :)

**Comment by andyvandesand** [comment]
Again to give you some background, I've just imported a year's worth of data. It might be easy enough to start from scratch and establish things bit by bit, but it's not that easy to make sense of things in this case.

**Comment by TheAxelander** [comment]
> Yes, all assigned, although still trying to get all transfers to add to 0.
> 
> Yes that screen is great, but for e.g. income up the top - currently not adding up, out by quite a significant amount, and there are only 5x transactions marked as income. But all the other categories at the top would be great to have some kind of explanation of how they are calculated, I'm just trying to work out what goes where but turns out it's a lot harder than anticipated :)

Ok yes that could be something that I could add to the documentation to share some more insights on how these figures come together. But did you noticed already that these figures are month-based, so you might need to navigate to the past to see some changes.

**Comment by TheAxelander** [comment]
> Again to give you some background, I've just imported a year's worth of data. It might be easy enough to start from scratch and establish things bit by bit, but it's not that easy to make sense of things in this case.

That's the way I also started. I started with an opening Income Transaction on a specific date (let's say 01.01.2020) with the Bank Balance on that day and then imported and assigned all upcoming Bank Transactions

**Comment by andyvandesand** [comment]
Yes I have noticed that. But even looking at just that month’s transactions I can’t quite make it work. But that’s cool, I’m working it out slowly – just making a suggestion on where some improvements might be great. Same with the InOut/Want/In/Activity columns, and “Distribute budget” button – it’s not quite clear straight up. I think I’m getting the idea, but it’s taking a lot of trial and error and tyring to look at the db to work out what I’m doing. 

Overall loving the product idea, and really hoping to be able to use it for my purposes. I’ve tried a lot of other options but don’t love the idea of having all my data in some cloud.

**Comment by TheAxelander** [comment]
I know, a lot of (wished) features evolved over the time and I can understand that not everything might be obvious at the first glance. As few days ago I started with a proper wiki-like documentation site I will definitely work on that.

Thanks a lot also for your feedback and input :)

**Comment by andyvandesand** [comment]
Yeah I can imagine! Keen to see if I can help out, I did a fair bit of c# and db stuff back in the days but it’s been a while…

Otherwise it might just have to be in beers next time I’m back in DE 😉

**Comment by andyvandesand** [comment]
Worked out the income issue. Is it possible that the app is taking all positive non-transfer transactions as income? I have classified a few reimbursement into various buckets, but they appear to be still counted as income. 
-------- Original message --------From: Alexander Preibisch ***@***.***> Date: 27/8/23  16:38  (GMT+08:00) To: TheAxelander/OpenBudgeteer ***@***.***> Cc: andyvandesand ***@***.***>, Author ***@***.***> Subject: Re: [TheAxelander/OpenBudgeteer] Require drill down capabilities for buckets (Issue #167) 
I know, a lot of (wished) features evolved over the time and I can understand that not everything might be obvious at the first glance. As few days ago I started with a proper wiki-like documentation site I will definitely work on that.
Thanks a lot also for your feedback and input :)

**Comment by TheAxelander** [comment]
Great to hear. So right now Income works in this way:

The Income value in the header on the Bucket Page includes all positive Transactions independently to which Bucket they are assigned, except to `Transfer` (this just prevents blown up values in case you move money between multiple accounts). Same would apply also to the Expenses value which also excludes everything from `Transfer`.

Let me share a Use Case why I think it should be possible to set positive Transactions to a certain Bucket:

Let's say I have a Bucket called `Games` where I put every month 30€ in. One month I buy a game on Steam for 10€, hence I have a expense with value `-10.00` which I assign to Bucket `Games`. The Balance is now at `20,00`.

While playing the new Game I decide to give it back and Steam makes a refund to my account. So now I have a transaction with `+10,00` which I again assign to Bucket `Games`. As a result my available budget which I can use for new games is back to `30,00`.

**Comment by andyvandesand** [comment]
Yes, but that’s my point exactly – not sure if we’re misunderstanding each other here, I completely agree, I _want_ to exclude positive transactions from income, but it doesn’t appear like I can do that right now?

 

I have classified a number of positive transactions into certain buckets, and some of them into the income bucket. Yet all of them are being counted towards income. This would make the income bucket almost superfluous right now, if the app just uses all positive transactions anyway (take away transfers)? I would suggest income should only be counting transactions categorised as income.

 

Separately, I would actually suggest to be able to provide a feature to further subcategorise income – in the previous app I used I distinguished between salary and other income, just in order to run reports on that later e.g. over the year.
